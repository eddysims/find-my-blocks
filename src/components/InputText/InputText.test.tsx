import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { InputText } from "./";

afterEach(cleanup);

it("render correctly", () => {
  const tree = renderer.create(<InputText defaultValue="hello" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a textarea if multiline is used", () => {
  const { getByPlaceholderText } = render(
    <InputText placeholder="placeholder" multiline />
  );
  const input = getByPlaceholderText("placeholder");
  expect(input).toBeInstanceOf(HTMLTextAreaElement);
});

it("calls the onchange function when typed in", () => {
  const placeholder = "Test";
  const value = "Hello";
  const changeHandler = jest.fn();
  const { getByPlaceholderText } = render(
    <InputText placeholder={placeholder} onChange={changeHandler} />
  );
  const input = getByPlaceholderText(placeholder);
  fireEvent.change(input, { target: { value: value } });

  expect(changeHandler).toHaveBeenCalledWith(value);
});
