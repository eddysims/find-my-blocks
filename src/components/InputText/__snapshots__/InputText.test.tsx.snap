// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <input
      className="components-text-control__input"
      id="inspector-text-control-0"
      onChange={[Function]}
      type="text"
      value="hello"
    />
  </div>
</div>
`;
